dockstore/dockstore
# Add enableAutoDois field to .dockstore.yml and allow disabling auto DOIs

**Description**
This PR reuses code from https://github.com/dockstore/dockstore/pull/5996. It adds an `enableAutoDois` boolean that can only be set to `false` to the .dockstore.yml so that auto DOI generation can be disabled for GitHub App workflowish entries. If present, the workflow will have it's auto DOI generation disabled and it will remain disabled unless we use the admin endpoint to enable it again. When we remove the admin endpoint, we'll want to allow the enabling of auto DOI generation through this property if the user changes their mind and wants it enabled again.

**Review Instructions**
Requires a hotfix deploy. In staging:
- Enable auto DOI generation for your published GitHub App workflow using the admin endpoint.
- Create a tag and verify that a DOI was automatically created.
- Add `enableAutoDois: false` to your workflow.
- Create a tag and verify that a DOI was NOT created.
- Try setting `enableAutoDois: true`. Check the GitHub App logs and verify that the push failed because `enableAutoDois` can only be set to false.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6805

**Security and Privacy**

If there are any concerns that require extra attention from the security team, highlight them here and check the box when complete. 


e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!
